### PR TITLE
[SR-14869][Sema] Don't record arg mismatch for function type mismatch at parameter

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Don't record apply argument mismatch fix for function type mismatch at parameter position of inout types. Skip that in `repairFailures` so we can report whole structural type mismatch.

Resolves SR-14869.

### CODE REVIEW

File: lib/Sema/CSSimplify.cpp [hunk 1]
```diff
@@ -4092,11 +4092,22 @@ bool ConstraintSystem::repairFailures(
...
+      if (path.back().is<LocatorPathElt::ContextualType>()) {
+        auto *locator = getConstraintLocator(anchor, path.back());
+        conversionsOrFixes.push_back(
+            IgnoreContextualType::create(*this, lhs, rhs, locator));
+        break;
+      }
+
+      // If this is a function type param type mismatch in any position,
+      // the mismatch we want to report is for the whole structural type.
+      while (path.back().is<LocatorPathElt::TupleElement>()) {
```
Comment (xedin) [inline lib/Sema/CSSimplify.cpp:4105]: Let's use an alternative approach and preserve `path` here for the fallthrough if it's not at function argument position - use rfind to determine where `FunctionArgument` is and then re-create `path` like `{path.begin(), path.begin() + <function arg position>}` or use `drop_back(size_t N)`?

**Review by xedin — COMMENTED** [review]
Thank you!

File: lib/Sema/CSSimplify.cpp (hunk 1 above)
Comment (LucianoPAlmeida) [reply lib/Sema/CSSimplify.cpp]: Makes sense, I just fixed =]

**Review by xedin — APPROVED** [review]

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test Linux Platform

**Pull request merged**